design/global-tag-throttling.md

```diff
@@ -3,44 +3,48 @@
 
-* Reserved read quota
-* Reserved write quota
-* Total read quota
-* Total write quota
+* Reserved quota
+* Total quota
 
-The global tag throttler can not throttle tags to a throughput below the reserved quotas, and it cannot allow throughput to exceed the total quotas.
+The global tag throttler can not throttle tags to a throughput below the reserved quota, and it cannot allow throughput to exceed the total quota.
 
 ### Cost
-The units for these quotas are computed as follows. The "cost" of a read operation is computed as:
+Internally, the units for these quotas are "page costs", computed as follows. The "page cost" of a read operation is computed as:
 
 ```
-readCost = bytesRead / SERVER_KNOBS->READ_COST_BYTE_FACTOR + 1;
+readCost = bytesRead / CLIENT_KNOBS->READ_COST_BYTE_FACTOR + 1;
 ```
 
-The "cost" of a write operation is computed as:
+The "page cost" of a write operation is computed as:
 
 ```
-writeCost = bytesWritten / CLIENT_KNOBS->WRITE_COST_BYTE_FACTOR + 1;
+writeCost = SERVER_KNOBS->GLOBAL_TAG_THROTTLING_RW_FUNGIBILITY_RATIO * ((bytesWritten / CLIENT_KNOBS->WRITE_COST_BYTE_FACTOR) + 1);
 ```
 
 Here `bytesWritten` includes cleared bytes. The size of range clears is estimated at commit time.
 
 ### Tuple Layer
-Tag quotas are stored inside of the system keyspace (with prefix `\xff/tagQuota/`). They are stored using the tuple layer, in a tuple of form: `(reservedReadQuota, totalReadQuota, reservedWriteQuota, totalWriteQuota)`. There is currently no custom code in the bindings for manipulating these system keys. However, in any language for which bindings are available, it is possible to use the tuple layer to manipulate tag quotas.
+Tag quotas are stored inside of the system keyspace (with prefix `\xff/tagQuota/`). They are stored using the tuple layer, in a tuple of form: `(reservedQuota, totalQuota)`. There is currently no custom code in the bindings for manipulating these system keys. However, in any language for which bindings are available, it is possible to use the tuple layer to manipulate tag quotas.
 
 ### fdbcli
-The easiest way for an external client to interact with tag quotas is through `fdbcli`. To get the quota of a particular tag, run the following command:
+The easiest way for an external client to interact with tag quotas is through `fdbcli`. To get the quota (in bytes/second) of a particular tag, run the following command:
 
 ```
-fdbcli> get <tag> [reserved|total] [read|write]
+fdbcli> quota get <tag> [reserved_throughput|total_throughput]
 ```
 
 To set the quota through `fdbcli`, run:
 
 ```
-fdbcli> set <tag> [reserved|total] [read|write] <value>
+fdbcli> quota set <tag> [reserved_throughput|total_throughput] <bytes_per_second>
+```
+
+Note that the quotas are specified in terms of bytes/second, and internally converted to page costs:
+
+```
+page_cost_quota = byte_quota / CLIENT_KNOBS->READ_COST_BYTE_FACTOR
```

Comment:
Should this include the +1?

```suggestion
page_cost_quota = byte_quota / CLIENT_KNOBS->READ_COST_BYTE_FACTOR + 1
```

Reply:
Looking at the implementation in fdbcli, I guess it doesn't apply the +1, but the comment farther above does suggest there is a +1, so I'll let you decide how to address that if necessary.

Reply:
Thanks for catching this, I resolved it by making the cost function round up to the nearest page size, which isn't exactly what they were doing previously.